chartfx-chart/src/main/java/de/gsi/chart/plugins/Screenshot.java

```diff
@@ -0,0 +1,139 @@
+package de.gsi.chart.plugins;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+
+import javafx.embed.swing.SwingFXUtils;
+import javafx.geometry.Insets;
+import javafx.geometry.Orientation;
+import javafx.scene.SnapshotParameters;
+import javafx.scene.control.Button;
+import javafx.scene.control.Separator;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.ClipboardContent;
+import javafx.scene.layout.HBox;
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
+
+import org.controlsfx.glyphfont.FontAwesome;
+import org.controlsfx.glyphfont.Glyph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Plugin allowing to take a screenshot of the complete chart.
+ * Allows to copy image to clipboard or to save as a file.
+ *
+ * @author Alexander Krimm
+ */
+public class Screenshot extends ChartPlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Screenshot.class);
+    private static final String FONT_AWESOME = "FontAwesome";
+    private static final int FONT_SIZE = 20;
+
+    private final HBox screenshotButtons = getScreenshotInteractorBar();
+
+    /**
+     * Create a screenshot plugin instance
+     */
+    public Screenshot() {
+        super();
+
+        chartProperty().addListener((change, o, n) -> {
+            if (o != null) {
+                o.getToolBar().getChildren().remove(screenshotButtons);
+            }
+            if (n != null) {
+                if (isAddButtonsToToolBar()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4880492)